src/nedap/utils/test/matchers.cljc

```diff
@@ -15,3 +19,25 @@
    :assert-expr-sym 'match?
    :pred-sym `matches?
    :pred =})
+
+(defrecord Gensym [expected]
+  matcher-combinators/Matcher
+  (-matcher-for [this] this)
+  (-matcher-for [this _] this)
+  (-match [_this actual]
+    (if (and (symbol? actual)
+             (= (str expected) (string/replace (str actual) #"(.?)(\d)*" "$1")))
```

Comment:
Not sure, but doesn't gensym _always_ add some digits? If so, shouldn't the `*` be a `+` in this regex?